src/Api/Serializer/RestSerializer.php

```diff
@@ -49,33 +50,28 @@ public function __construct(Service $api, $endpoint)
      */
     public function __invoke(
         CommandInterface $command,
-        $endpointProvider = null,
-        $clientArgs = null
+        $endpoint = null
     )
     {
         $operation = $this->api->getOperation($command->getName());
         $commandArgs = $command->toArray();
         $opts = $this->serialize($operation, $commandArgs);
-        $headers = isset($opts['headers']) ? $opts['headers'] : [];
-
-        if ($endpointProvider instanceof EndpointProviderV2) {
-            $this->setRequestOptions(
-                $endpointProvider,
-                $command,
-                $operation,
-                $commandArgs,
-                $clientArgs,
-                $headers
-            );
-            $this->endpoint = new Uri($this->endpoint);
+        $headers = $opts['headers'] ?? [];
+
+        if ($endpoint instanceof RulesetEndpoint) {
```

Comment:
This code appears to be the same for all the serializers; is there a reason it's not in a method on the EndpointV2SerializerTrait?